### DISCUSSION TITLE
android上播放bmp预合成的pag文件，会有大量ccodec日志打印

### DISCUSSION THREAD

**Discussion by shenxing118** [Tencent/libpag#1769] [Q&A] [answered]
android上播放bmp预合成的pag文件，会有大量ccodec日志打印，但不同分辨率表现不一样，以下zip包中的两个文件只有分辨率的区别，分辨率分别为240*240和360*360，240不会打印ccodec日志，360的会打印，请问是有不同的策略吗
CCodecBufferChannel     com...t.libpag.sample.libpag_sample  D  [c2.qti.avc.decoder#327] Query input allocators returned 0 params => BAD_INDEX (6)
C2Store                 com...t.libpag.sample.libpag_sample  D  Using ION
CCodecBufferChannel     com...t.libpag.sample.libpag_sample  D  [c2.qti.avc.decoder#327] Created input block pool with allocatorID 16 => poolID 17 - OK (0)
CCodecBufferChannel     com...t.libpag.sample.libpag_sample  D  [c2.qti.avc.decoder#327] Query output allocators returned 0 params => BAD_INDEX (6)
CCodecBufferChannel     com...t.libpag.sample.libpag_sample  D  [c2.qti.avc.decoder#327] Query output surface allocator returned 0 params => BAD_INDEX (6)
CCodecBufferChannel     com...t.libpag.sample.libpag_sample  I  [c2.qti.avc.decoder#327] Created output block pool with allocatorID 18 => poolID 54368 - OK
CCodecBufferChannel     com...t.libpag.sample.libpag_sample  D  [c2.qti.avc.decoder#327] Configured output block pool ids 54368 => OK
Codec2-Out...ufferQueue com...t.libpag.sample.libpag_sample  D  remote graphic buffer migration 0/0
Codec2Client            com...t.libpag.sample.libpag_sample  D  setOutputSurface -- failed to set consumer usage (6/BAD_INDEX)
Codec2Client            com...t.libpag.sample.libpag_sample  D  setOutputSurface -- generation=27419649 consumer usage=0x900
Codec2Client            com...t.libpag.sample.libpag_sample  D  Surface configure completed
ion                     com...t.libpag.sample.libpag_sample  E  ioctl c0044901 failed with code -1: Inappropriate ioctl for device
CCodecConfig            com...t.libpag.sample.libpag_sample  D  c2 config diff is   c2::u32 output.delay.value = 10
Codec2-Out...ufferQueue com...t.libpag.sample.libpag_sample  D  set max dequeue count 17 from update
CCodecConfig            com...t.libpag.sample.libpag_sample  D  c2 config diff is   c2::u32 output.delay.value = 18
Codec2-Out...ufferQueue com...t.libpag.sample.libpag_sample  D  set max dequeue count 25 from update
CCodecConfig            com...t.libpag.sample.libpag_sample  D  c2 config diff is   c2::u32 raw.color.matrix = 3
                                                                c2::u32 raw.color.primaries = 4
                                                                c2::u32 raw.color.range = 2
                                                                c2::u32 raw.color.transfer = 3
                                                                c2::u32 raw.crop.height = 596
                                                                c2::u32 raw.crop.left = 0
                                                                c2::u32 raw.crop.top = 0
                                                                c2::u32 raw.crop.width = 304
[pag文件.zip](https://github.com/Tencent/libpag/files/12855735/pag.zip)

**Comment by domchen** [comment, marked as answer]
这个日志不是PAG打的，Android系统用硬解的时候会正常输出一些信息，分辨率比较小的时候会采用软件解码速度更快，不走硬解可能就没日志了。
